x/ibc/core/client/query.go

```diff
@@ -18,20 +18,24 @@ import (
 // to perform the query should be set in the client context. The query will be
 // performed at one below this height (at the IAVL version) in order to obtain
 // the correct merkle proof. Proof queries at height less than or equal to 2 are
-// not supported.
+// not supported. Queries with a client context height of 0 will perform a query
+// at the lastest state available.
 // Issue: https://github.com/cosmos/cosmos-sdk/issues/6567
 func QueryTendermintProof(clientCtx client.Context, key []byte) ([]byte, []byte, clienttypes.Height, error) {
 	height := clientCtx.Height
 
 	// ABCI queries at height less than or equal to 2 are not supported.
 	// Base app does not support queries for height less than or equal to 1.
 	// Therefore, a query at height 2 would be equivalent to a query at height 3
-	if clientCtx.Height <= 2 {
+	if height != 0 && height <= 2 {
```

Comment:
turns out, somewhere down the line, a height of 0 is interpreted as latest height

Reply:
can you update the comment too?

Reply:
updated, I left the error message as is